### ISSUE TITLE
Buffer is always null

### ISSUE THREAD

**Issue by sayem314** [zulhilmizainuddin/multipart-download#10]
Code example

```
const MultipartDownload = require('multipart-download');

const multipartDownload = (url) => {
  return new Promise((resolve, reject) => {
    new MultipartDownload()
      .start(url, {
        numOfConnections: 4,
        writeToBuffer: true,
      })
      .on('error', (err) => {
        console.log(err);
        reject(err);
      })
      .on('end', (buffer) => {
        console.log(buffer);
        resolve(buffer);
      });
  });
};

module.exports = multipartDownload;
```

**Comment by sayem314** [comment]
Another example:

```
$ cat test.js
const MultipartDownload = require('multipart-download');

new MultipartDownload()
  .start('https://sample-videos.com/img/Sample-jpg-image-50kb.jpg', {
    numOfConnections: 5,
    writeToBuffer: true,
  })
  .on('error', (err) => {
    // handle error here
  })
  .on('data', (data, offset) => {
    // manipulate data here
  })
  .on('end', (buffer) => {
    console.log(Buffer.isBuffer(buffer));
    console.log(`Downloaded file buffer: ${buffer}`);
  });
```

**Comment by zulhilmizainuddin** [comment]
Fixed in v1.2.2
https://www.npmjs.com/package/multipart-download/v/1.2.2
